cmd/bnsd/scenarios/main_test.go

```diff
@@ -91,8 +92,17 @@ func TestMain(m *testing.M) {
 	// run the app inside a tendermint instance
 	node = rpctest.StartTendermint(app)
 	bnsClient = client.NewClient(client.NewLocalConnection(node))
-	time.Sleep(startupDelay) // wait for chain
-	code := m.Run()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+	_, err = weaveClient.NewClient(weaveClient.NewLocalConnection(node)).WaitForNextBlock(ctx)
```

Comment:
Question: this usage seems clean enough, but a little verbose. Do you have any concrete suggestions to make this simpler? Maybe combine NewClient and NewLocalConnection into NewLocalClient??

Reply:
Yes, I looked at that and I was even thinking to have a helper of some sort in client lib to reuse here too. I'll create a ticket for that